src/main/java/org/opensearch/ad/AnomalyDetectorExtension.java

```diff
@@ -145,7 +145,9 @@ public List<ExtensionRestHandler> getExtensionRestHandlers() {
                 new RestIndexAnomalyDetectorAction(extensionsRunner(), restClient(), javaAsyncClient()),
                 new RestValidateAnomalyDetectorAction(extensionsRunner(), restClient(), javaAsyncClient()),
                 new RestGetAnomalyDetectorAction(extensionsRunner(), restClient()),
-                new RestAnomalyDetectorJobAction(extensionsRunner(), restClient())
+                new RestAnomalyDetectorJobAction(extensionsRunner(), restClient()),
+                new RestStatsAnomalyDetectorAction(extensionsRunner(), restClient(), adStats, nodeFilter)
```

Comment:
ADStats and DiscoveryNodeFilterer objects are already returned via `createComponents()` so they are available by injection in the action.

Reply:
ACK

Reply:
It is a rest action so I cannot inject the required dependency via guice. If it would have been a transport Action then I could have achieved the dependency injection.